centreon/src/App/MonitoringConfiguration/Infrastructure/FileSystem/FileSystemPluginRepository.php

```diff
@@ -46,10 +51,41 @@ public function findByPath(OptionValue $path): Collection
 
         return new Collection(
             array_map(
-                static fn (SplFileInfo $plugin): Plugin => new Plugin(new PluginName($plugin->getFilename())),
+                static fn (SplFileInfo $plugin): Plugin => new Plugin(
+                    name: new PluginName($plugin->getFilename()),
+                    commandLine: new PluginCommandLine($plugin->getRealPath()),
+                ),
                 iterator_to_array($pluginInfos->getIterator())
             ),
             Plugin::class
         );
     }
+
+    public function getByPathAndName(OptionValue $path, PluginName $name): Plugin
+    {
+        $pluginInfos = $this->finder->files()
+            ->in($path->value)
+            ->name($name->value);
+
+        if (! $this->finder->hasResults()) {
+            throw new PluginNotFoundException(['path' => $path->value, 'name' => $name->value]);
+        }
+
+        $plugin = current(iterator_to_array($pluginInfos));
```

Comment:
```suggestion
        $pluginInfos->rewind();
        $plugin = $pluginInfos->current();
```
To save up some memory

Reply:
this does not work :/ , I'll keep this code as the memory cost is not high, I'll merge like this as the feature is near the end, but we can discuss about it